utils/firestore.js

```diff
@@ -1799,6 +1841,7 @@ const getNotificationSpecifications = async (notificationType, notificationCateg
     }
 }
 
+// TODO: Avoid using `offset` for pagination, because offset documents are still read and charged.
```

Comment:
can you explain this? Example, say I have 1000 documents and my limit is 100. If I'm using `offset` for pagination, are you saying we will in the end be charged for reading more than 1000 documents?

Reply:
I believe so: the offset docs are also counted and charged. I'm now writing based on what I understand, so the below data may not be accurate.
In the example mentioned, if pagination using `offset`  to get all the 1000 documents and 100 docs retrieved per pagination, the total count of reads will be 100 + 200 + 300 +.. + 1000 = 5500.

Reply:
This is something definitely worth looking into / verifying... We use offset for pagination in our APIs that return the largest amount of data, such as `getParticipants`, if what you're saying is correct then it's adding way more than we expect to our read document count.

Reply:
Here is documentation explaining why `offset` should be avoided when reading large datasets: https://firebase.google.com/docs/firestore/pricing